src/networks/invertible_network_conditional_hint_multiscale.jl

```diff
@@ -54,21 +54,23 @@ export NetworkMultiScaleConditionalHINT, NetworkMultiScaleConditionalHINT3D
 
  See also: [`ActNorm`](@ref), [`ConditionalLayerHINT!`](@ref), [`get_params`](@ref), [`clear_grad!`](@ref)
 """
-struct NetworkMultiScaleConditionalHINT <: InvertibleNetwork
+mutable struct NetworkMultiScaleConditionalHINT <: InvertibleNetwork
```

Comment:
I just saw that and in other places so I guess I'll have to go through the code. `mutable` is in general a bad idea (fabio would kill anyone who uses mutable objects), if something states change it should be a new object. So that `tag_as_reversed`  is not very good. Not your fault, just following what's there, so I guess comment for myself to fix.

Reply:
Added some tests. I am not passing gradient test for reversed multi scale. I will come back later to figure out why. 
```
Gradient test cond. HINT net: input
returned 458.55124; 16.934978
returned 433.00421; 4.7388973
returned 417.610535; 1.2610188
returned 49.134384; 0.30139256
returned 44.640869; 0.077019215
returned 42.3399506; 0.018993616
2
Invertibility test HINT network
Test Failed at REPL[23]:11
  Expression: isapprox(norm(X - X_) / norm(X), 0.0f0; atol = 0.001f0)
   Evaluated: isapprox(0.0019741387f0, 0.0f0; atol = 0.001f0)
ERROR: There was an error during testing
```

Reply:
I spend some time on this and cant find the error. Could I please get another pair of eyes on the gradient test for a reversed multiscale conditional network?

Reply:
Tests pass now.

Reply:
:+1: